azure/test.yml

```diff
@@ -24,33 +25,38 @@ jobs:
           vmImage: macOS-10.14
         Windows:
           vmImage: windows-2019
-  ${{ if ne('true', parameters.cross) }}:
-    variables:
+  variables:
+    ${{ if ne('true', parameters.cross) }}:
       vmImage: ubuntu-16.04
+    options: ''
   pool:
     vmImage: $(vmImage)
   steps:
   - template: _setup.yml
     parameters:
       rust: ${{ parameters.rust }}
       setup: ${{ parameters.setup }}
-  - ${{ if ne('true', parameters.single_threaded) }}:
-    - script: cargo test --all --features "${{ parameters.features }}"
-      displayName: Run tests
-      env:
-        ${{ insert }}: ${{ parameters.envs }}
-  - ${{ if eq('true', parameters.single_threaded) }}:
-    - script: cargo test --all --features "${{ parameters.features }}" -- --test-threads=1
-      displayName: Run tests (single-threaded)
+  # Cargo options go here
+  - script: echo '##vso[task.setvariable variable=options]$(options) --all'
+  - ${{ if eq(true, parameters.features)}}:
+    - script: echo '##vso[task.setvariable variable=options]$(options) --features "${{ parameters.features }}"'
+  # Add any additional conditional libtest options to this check
+  - ${{ if or(eq(true, parameters.single_threaded), eq(true, parameters.test_ignored))}}:
```

Comment:
This won't work, as any non-empty string coerces to `true`. This would mean that if someone passes:
```yaml
parameters:
  test_ignored: 'false'
```
(which some of the Azure DevOps docs recommend even though they shouldn't), then that will be considered `true`. Comparing against the string does not have this issue, as booleans are coerced to the appropriate string (`'true'` and `'false'`).

Reply:
Fair enough, I was getting strange results earlier when comparing strings against strings, which I why I changed it

Reply:
I _think_  the trick here is to use`eq('true', parameters.x)` and `ne('true', parameters.x)` for parameters you _expect_ people to set to `true` if they set them, and `eq('false', parameters.x)` and `ne('false', parameters.x)` for parameters you expect people to set to false.

Reply:
See also https://github.com/crate-ci/azure-pipelines/pull/50

Reply:
I've corrected all the checks which expect a bool, but might get a string. Hopefully patching this up a little